frontend/controller/serviceworkers/sw-primary.js

```diff
@@ -325,7 +325,7 @@ const setupPromise = setupChelonia()
 
 self.addEventListener('install', function (event) {
   console.debug('[sw] install')
-  event.waitUntil(Promise.all([setupPromise, self.skipWaiting()]))
+  event.waitUntil(setupPromise.then(() => self.skipWaiting()))
```

Comment:
Are you sure it shouldn't be `skipWaiting.then(() => self.setupPromise())` ?

(I'm not saying it should be... I just don't know myself)

Reply:
Not sure myself, and the documentation isn't very helpful here. However, I think this is correct. `skipWaiting` tells the browser that the SW should immediately move from `installed` to `activated` (or from `waiting` to `activated`). We only want to do this after setup is complete, so this seems reasonable. With the previous code, I think it worked the same way too, since it'd wait until `setupPromise` had resolved, but now it's more explicit.

TL;DR: I think it doesn't matter when `skipWaiting` is called, but, if it made a difference, what's now there seems more correct.

Reply:
Can you copy/paste that as a comment into the code?